### PR TITLE
JST-598: simplify worker type

### PR DESCRIPTION
**Pull request by SewerynKras**
BREAKING CHANGE: worker type no longer takes two parameters

### CODE REVIEW

File: src/executor/executor.ts [hunk 1]
```diff
@@ -117,7 +117,7 @@ export class TaskExecutor {
   private networkService?: NetworkService;
   private statsService: StatsService;
   private activityReadySetupFunctions: Worker[] = [];
-  private taskQueue: TaskQueue<Task<unknown, unknown>>;
+  private taskQueue: TaskQueue<Task>;
```
Comment (grisha87) [inline src/executor/executor.ts:120]: One nice improvement can be to change the signature to `TaskQueue<T extends QueueableTask = Task>` to provide a nice default Then we could just use `private taskQueue: TaskQueue;` instead.

File: src/golem_network/golem_network.ts [hunk 2]
```diff
@@ -98,7 +98,7 @@ export class GolemNetwork {
    * console.log(status);
    * ```
    */
-  public async createJob<Output = unknown>(worker: Worker<unknown, Output>) {
+  public async createJob<Output = unknown>(worker: Worker<Output>) {
```
Comment (grisha87) [inline src/golem_network/golem_network.ts:101]: I think that we could drop `= unknown` as the type inference would work here just perfectly.

File: tests/unit/tasks_service.test.ts [hunk 3]
```diff
@@ -2,18 +2,18 @@ import * as activityMock from "../mock/rest/activity";
...
     logger.clear();
     activityMock.clear();
-    queue = new TaskQueue<Task<any, any>>();
+    queue = new TaskQueue<Task>();
   });
   it("should process new task in queue", async () => {
-    const worker: Worker<null, Result> = async (ctx) => ctx.run("some_shell_command");
-    const task = new Task<null, Result>("1", worker);
+    const worker: Worker<Result> = async (ctx) => ctx.run("some_shell_command");
+    const task = new Task<Result>("1", worker);
```
Comment (grisha87) [inline tests/unit/tasks_service.test.ts:16]: Suggestion: Write code to promote type inference. The equivalent would be:

```ts
const worker = async (ctx: WorkContext) => ctx.run("some_shell_command");
const task = new Task("1", worker);
```

This way We don't have to specify `Worker<Result>` | `Task<Result>` and the type inference does the job.

File: tests/unit/work.test.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@ describe("Work Context", () => {
   describe("Executing", () => {
     it("should execute run command", async () => {
       const activity = await Activity.create("test_agreement_id", yagnaApi);
-      const worker: Worker<void, Result> = async (ctx) => ctx.run("some_shell_command");
+      const worker: Worker<Result> = async (ctx) => ctx.run("some_shell_command");
```
Comment (grisha87) [inline tests/unit/work.test.ts:20]: Same comment about type inference applies here. Refactored example:

```ts
const worker = async (ctx: WorkContext) => ctx.run("some_shell_command");
```

**Review by grisha87 — APPROVED** [review]

**Comment by SewerynKras** [comment]
With the change to `executor.run`'s return type, we now get the correct type inference:
image

For reference, previously the type union always had `| undefined`
image

**Review by grisha87 — APPROVED** [review]

**Pull request merged**